crates/cdk/src/nuts/nut04.rs

```diff
@@ -81,9 +84,10 @@ impl FromStr for QuoteState {
 /// Mint quote response [NUT-04]
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[cfg_attr(feature = "swagger", derive(utoipa::ToSchema))]
-pub struct MintQuoteBolt11Response {
+#[serde(bound = "Q: Serialize + DeserializeOwned")]
+pub struct MintQuoteBolt11Response<Q> {
     /// Quote Id
-    pub quote: String,
+    pub quote: Q,
```

Comment:
> Having MeltId wrap an String or an UUID based on a cfg.

I don't think we can do it based on cfg as features are additive so both the `mint` and `wallet` feature could be and are by default enabled at the same time.

Reply:
We could also keep it as a string in the Response as to be inline with the nut. And then use the uuid in only the mint mod.

Reply:
`type MintQuoteBolt11Response = nut04::MintQuoteBolt11Response<Uuid>`
This is just an alias, syntax sugar, and doesn't change anything in terms of having or not a generic, it's just maybe easier to read. 

Is readability your only issue?

Reply:
> We could also keep it as a string in the Response as to be inline with the nut. And then use the uuid in only the mint mod.

Doesn't really make sense IMO.
It happens that the protocol specifies that the id can be anything that can be serialized into a string.
It happens that in some part of the codebase, we know for sure it is a Uuid, and in some other parts we don't know so we go for the smaller common denominator, a `String`.

But going from bytes to String to Uuid would add an intermediate, non-necessary step in the serialization.

I don't really see any way to avoid a generic here

Reply:
Let's merge it that way and see if it becomes a problem for users.
I'm usually in favor of being explicit until it becomes an overcharge of information.
Do you agree?

Reply:
I'm happy with it as is.